### PR TITLE
[UICIRC-91] Use documented react-intl patterns instead of stripes.intl

### PR DESCRIPTION
**Pull request by anastasiaHahanova**
## Purpose
Taking into account that retrieving the intl object from this.context.intl or the stripes god object (stripes.intl) is a deprecated pattern.

The purpose of this PR is to update this repo to only use react-intl constructs outlined at https://github.com/folio-org/stripes-core/blob/master/doc/i18n.md

## Learning
React Intl docs: https://github.com/yahoo/react-intl/wiki

### CODE REVIEW

File: settings/FixedDueDateScheduleDetail.js [hunk 1]
```diff
@@ -69,9 +72,9 @@ class FixedDueDateScheduleDetail extends React.Component {
             </Col>
           </Row>
           <Row key={index + 2}>
-            <Col xs={4}>{this.formatDate(schedule.from)}</Col>
-            <Col xs={4}>{this.formatDate(schedule.to)}</Col>
-            <Col xs={4}>{this.formatDate(schedule.due)}</Col>
+            <Col xs={4}>{formatDate(schedule.from)}</Col>
```
Comment (Godlevskyi) [inline settings/FixedDueDateScheduleDetail.js]: Please use \<FormattedDate\> component here

File: settings/FixedDueDateScheduleDetail.js [hunk 2]
```diff
@@ -44,8 +44,11 @@ class FixedDueDateScheduleDetail extends React.Component {
   }
 
   render() {
-    const fixedDueDateSchedule = this.props.initialValues;
-    const formatMsg = this.props.stripes.intl.formatMessage;
+    const {
+      intl: { formatDate, formatMessage },
```
Comment (Godlevskyi) [inline settings/FixedDueDateScheduleDetail.js]: No need to use formatDate, please import <formatedDate> component

File: settings/FixedDueDateScheduleForm.js [hunk 3]
```diff
@@ -143,28 +154,34 @@ Delete
   }
 
   renderPaneTitle() {
-    const { initialValues, stripes: { intl } } = this.props;
+    const {
+      initialValues,
```
Comment (Godlevskyi) [inline settings/FixedDueDateScheduleForm.js]: InitialValues direcly reassigned after destructuring. So, it make sence to reassign in the destructuring and  also set default value

File: settings/FixedDueDateScheduleForm.js [hunk 4]
```diff
@@ -143,28 +154,34 @@ Delete
...
+      intl: { formatMessage },
+    } = this.props;
+
     const selectedSet = initialValues || {};
```
Comment (Godlevskyi) [inline settings/FixedDueDateScheduleForm.js]: Should be removed. Se comment above

File: settings/LoanPolicyForm.js [hunk 5]
```diff
@@ -50,7 +51,7 @@ class LoanPolicyForm extends React.Component {
   }
 
   getCurrentValues() {
-    const { stripes: { store } } = this.props;
+    const { store } = this.props.stripes;
```
Comment (Godlevskyi) [inline settings/LoanPolicyForm.js]: What is the reason for that change?

File: settings/LoanRules.js [hunk 6]
```diff
@@ -21,6 +22,7 @@ const editorDefaultProps = {
     m: 'Material Type',
     t: 'Loan Type',
   },
+  intl: intlShape.isRequired,
```
Comment (Godlevskyi) [inline settings/LoanRules.js]: ?? what for it is nat a propTypes

File: settings/LoanRules.js [hunk 7]
```diff
@@ -162,7 +180,12 @@ class LoanRules extends React.Component {
   }
 
   render() {
-    if (!this.props.resources.loanTypes) {
+    const {
+      resources,
+      intl: { formatMessage },
+    } = this.props;
+
+    if (!resources.loanTypes) {
```
Comment (Godlevskyi) [inline settings/LoanRules.js]: Only loanTypes used from resources, so it could be movewd to destructuring above, like:
resources: { loanTypes }

File: settings/StaffSlips/PreviewModal.js [hunk 8]
```diff
@@ -39,9 +39,15 @@ class PreviewModal extends React.Component {
...
-    const closeLabel = intl.formatMessage({ id: 'ui-circulation.settings.staffSlips.close' });
-    const heading = intl.formatMessage({ id: 'ui-circulation.settings.staffSlips.previewLabel' });
+    const {
+      open,
+      onClose,
+      previewTemplate,
+      intl: { formatMessage },
+    } = this.props;
+
+    const closeLabel = formatMessage({ id: 'ui-circulation.settings.staffSlips.close' });
```
Comment (Godlevskyi) [inline settings/StaffSlips/PreviewModal.js]: Please use \<FormatedMessage\> here as string is not required

File: settings/LoanPolicyForm.js (hunk 5 above)
Comment (anastasiaHahanova) [reply settings/LoanPolicyForm.js]: I see no need for nested destructuring in this case. Unlikely some other properties of props will be used in this method, therefore, now it looks more readable.

File: settings/FixedDueDateScheduleForm.js [hunk 9]
```diff
@@ -143,28 +154,32 @@ Delete
   }
 
   renderPaneTitle() {
-    const { initialValues, stripes: { intl } } = this.props;
-    const selectedSet = initialValues || {};
+    const {
+      initialValues: selectedSet = {},
```
Comment (mkuklis) [inline settings/FixedDueDateScheduleForm.js]: I think this will override the `initialValues` which changes the previous implementation. Could you please change it back to what it was? or just do:

`const selectedSet = this.props.initialValues || {}`

We have to be careful not to mutate `initialValues`. redux-form is sensitive about that object.

File: settings/StaffSlips/StaffSlipDetail.js [hunk 10]
```diff
@@ -52,7 +53,7 @@ class StaffSlipDetail extends React.Component {
 
   render() {
     const { openDialog } = this.state;
-    const staffSlip = this.props.initialValues;
+    const { initialValues: staffSlip } = this.props;
```
Comment (mkuklis) [inline settings/StaffSlips/StaffSlipDetail.js]: This also can't be right. Could you please revert this change?

File: settings/FixedDueDateScheduleForm.js [hunk 11]
```diff
@@ -109,24 +114,30 @@ class FixedDueDateScheduleForm extends React.Component {
...
             <Button
               id="clickable-delete-set"
-              title={intl.formatMessage({ id: 'ui-circulation.settings.fDDSform.delete' })}
+              title={formatMessage({ id: 'ui-circulation.settings.fDDSform.delete' })}
               buttonStyle="danger"
               onClick={this.beginDelete}
               disabled={confirmDelete}
             >
-Delete
+              Delete
```
Comment (mkuklis) [inline settings/FixedDueDateScheduleForm.js]: @anastasiaHahanova would you mind also translating this one? Thank you!

File: settings/FixedDueDateScheduleForm.js (hunk 9 above)
Comment (anastasiaHahanova) [reply settings/FixedDueDateScheduleForm.js]: Hi Michał, you probably haven't mentioned that in this particular case I have applied ES6 object destructuring pattern. You can have a look of the docs [here](https://javascript.info/destructuring-assignment). In this case I believe we don't need to create a new variable as we can distructure it directly from this.props object with the default value.

Comment (mkuklis) [reply settings/FixedDueDateScheduleForm.js]: Oh you are right @anastasiaHahanova! Sorry about that. Somehow it was hard spot that for me.

File: settings/StaffSlips/StaffSlipDetail.js (hunk 10 above)
Comment (mkuklis) [reply settings/StaffSlips/StaffSlipDetail.js]: This of course is also fine.
